app/components/LabelWithLink.tsx

```diff
@@ -12,22 +12,22 @@ import { useEnsName } from 'wagmi';
 
 import { AiAgent } from '@/types';
 
-interface AddressLinkProps extends ComponentProps<typeof Link> {
+interface LabelWithLinkProps extends ComponentProps<typeof Link> {
   address: Address;
   aiAgent?: AiAgent;
   className?: string;
   href: string;
   iconSize?: number;
 }
 
-export const AddressLink = ({
+export const LabelWithLink = ({
```

Comment:
Why not remove With? I think it's a label and a link

Reply:
it's a bit of logic to handle the "ai" label, we used it in 2 places, so let's keep for now.